src/huggingface_hub/hf_api.py

```diff
@@ -742,19 +742,19 @@ def create_repo(
             raise ValueError("Invalid repo type")
 
         json = {"name": name, "organization": organization, "private": private}
-        if repo_type is not None:
+        if repo_type is not None and repo_type != "model":
```

Comment:
on the server side you can probably pass type="model" to create a model repo

(so you can do `json["type"] = repo_type` also in that case)

Reply:
So we're saying just get rid of the `None` entirely and just force the user to do explicitly `model`? (And set that as a default)

Reply:
no (i mean, not necessarily), just that you can remove the `and repo_type != "model" ` clause if you want

But if it's simpler yeah you can also get rid of `None` (i'd try to staying backward compatible if possible though)

Reply:
Yes, staying backward compatible is super important, indeed

Reply:
Got it, okay. Just making sure. Didn't think we wanted to go down that path